source/plugins/ruby/in_kube_podmdminventory.rb

```diff
@@ -186,20 +186,24 @@ def getMDMRecords()
       retryAttemptCount = 1
       mdmRecords = {}
       begin
-        f = File.open(Constants::MDM_POD_INVENTORY_STATE_FILE, "r")
-        if !f.nil?
-          isAcquiredLock = f.flock(File::LOCK_EX | File::LOCK_NB)
-          raise "in_kube_podmdminventory:getMDMRecords:Failed to acquire file lock @ #{Time.now.utc.iso8601}" if !isAcquiredLock
-          startTime = (Time.now.to_f * 1000).to_i
-          mdmRecords = JSON.parse(f.read)
-          timetakenMs = ((Time.now.to_f * 1000).to_i - startTime)
-          if mdmRecords.nil? || mdmRecords.empty? || mdmRecords["items"].nil? || mdmRecords["collectionTime"] == @prevCollectionTime
-            raise "in_kube_podmdminventory:getMDMRecords: either read mdmRecords is nil or empty or stale @ #{Time.now.utc.iso8601}"
+        if File.exist?(Constants::MDM_POD_INVENTORY_STATE_FILE)
+          f = File.open(Constants::MDM_POD_INVENTORY_STATE_FILE, "r")
+          if !f.nil?
+            isAcquiredLock = f.flock(File::LOCK_EX | File::LOCK_NB)
+            raise "in_kube_podmdminventory:getMDMRecords:Failed to acquire file lock @ #{Time.now.utc.iso8601}" if !isAcquiredLock
+            startTime = (Time.now.to_f * 1000).to_i
+            mdmRecords = JSON.parse(f.read)
+            timetakenMs = ((Time.now.to_f * 1000).to_i - startTime)
+            if mdmRecords.nil? || mdmRecords.empty? || mdmRecords["items"].nil? || mdmRecords["collectionTime"] == @prevCollectionTime
+              raise "in_kube_podmdminventory:getMDMRecords: either read mdmRecords is nil or empty or stale @ #{Time.now.utc.iso8601}"
+            end
+            @prevCollectionTime = mdmRecords["collectionTime"]
+            $log.info "in_kube_podmdminventory:getMDMRecords:Number of MDM records: #{mdmRecords["items"].length} with time taken(ms) for read: #{timetakenMs} @ #{Time.now.utc.iso8601}"
+          else
+            raise "in_kube_podmdminventory:getMDMRecords:Failed to open file for read @ #{Time.now.utc.iso8601}"
           end
-          @prevCollectionTime = mdmRecords["collectionTime"]
-          $log.info "in_kube_podmdminventory:getMDMRecords:Number of MDM records: #{mdmRecords["items"].length} with time taken(ms) for read: #{timetakenMs} @ #{Time.now.utc.iso8601}"
         else
-          raise "in_kube_podmdminventory:getMDMRecords:Failed to open file for read @ #{Time.now.utc.iso8601}"
+          $log.warn "in_kube_podmdminventory:getMDMRecords:File does not exist: #{Constants::MDM_POD_INVENTORY_STATE_FILE} @ #{Time.now.utc.iso8601}"
```

Comment:
same here . the change to remove the  line  ApplicationInsightsUtility.sendExceptionTelemetry(err) code at https://github.com/microsoft/Docker-Provider/blob/ci_prod/source/plugins/ruby/in_kube_podinventory.rb#L296